sdk/core/azure-core-test/src/main/java/com/azure/core/test/utils/TestUtils.java

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.test.utils;
+
+import org.junit.jupiter.api.Assertions;
+
+import java.nio.ByteBuffer;
+import java.util.Arrays;
+import java.util.Objects;
+
+/**
+ * Contains utility methods used for testing.
+ */
+public final class TestUtils {
+    /**
+     * Asserts that two arrays are equal.
+     * <p>
+     * This method is similar to JUnit's {@link Assertions#assertArrayEquals(byte[], byte[])} except that it takes
+     * advantage of hardware intrinsics offered by the JDK to optimize comparing the byte arrays.
+     * <p>
+     * If the arrays aren't equal this will call {@link Assertions#assertArrayEquals(byte[], byte[])} to take advantage
+     * of the better error message, but this is the exceptional case and worth the double comparison performance hit.
+     *
+     * @param expected The expected byte array.
+     * @param actual The actual byte array.
+     */
+    public static void assertArraysEqual(byte[] expected, byte[] actual) {
+        if (!Arrays.equals(expected, actual)) {
+            assertArraysEqual(expected, actual);
+        }
```

Comment:
We can use `Arrays.mismatch()` to find the index of the first byte mismatch. We can then log the error message with the index. We don't need to do the double comparison.

Note: `Arrays.mismatch()` is also used by `Arrays.equals()` internally.

Reply:
`Arrays.mismatch()` is only available in Java 9+ (https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/Arrays.html#mismatch(byte%5B%5D,byte%5B%5D))